lib/Controller.php

```diff
@@ -112,10 +112,12 @@ class Controller
      *
      * initializes and runs PrivateBin
      *
+     * @param ?Configuration $config
+     *
      * @access public
      * @throws Exception
      */
-    public function __construct()
+    public function __construct(?Configuration $config = null)
```

Comment:
Just FYI: Interesting, cool new syntax did not know this yet?

May I ask what this is? Quickly searched for it here but did not found it: https://stackoverflow.com/questions/28395855/how-to-create-constructor-with-optional-parameters

Reply:
I used the nullable type! https://www.php.net/manual/en/migration71.new-features.php

Using the `Configuration|null` type would have been ok but it's not compatible with PHP < 8.2.

Reply:
Should work as we now require PHP 7.3, but should it fail in one of our version specific unit tests (which only run on our project internal branches) we could still revert to a more compatible syntax. Let's try and make use of it.